[#187364719] GitHub actions with readme badge

Bananayosostene opened the pull request:
Integration GitHub actions with readme badge 

# Why is this important

This is required to automatically run tests and other preliminary build steps required to ship the next iteration of the product

# Setting up continuous integration

munezerobagira commented:
can you check your title

AngeloChristian1 commented:
Add a pivotal id on the name of the PR

Yvan-David commented on .github/workflows/node.js.yml:
```diff
@@ -0,0 +1,37 @@
+name: Node.js CI
+
+on:
+  push:
```
it should be configured to develop branch or any branch so that it doesn't depend on your branch

munezerobagira commented on src/math.js:
what is role of this js file?
It seem to be a build version of ts
can you remove it?

munezerobagira commented on src/migrations/20240407175814-create-user.js:
**normally** deleting migration file is not recommended

munezerobagira commented on test/math.test.js:
```diff
@@ -0,0 +1,18 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const chai_1 = require("chai");
+const math_1 = require("../src/math");
+describe("Testing Index", () => {
+    it("should return 10 when have 8 + 2", () => {
```
this also seem to be a build version of ts file,
can you remove it too?

munezerobagira reviewed (changes requested):
can  you include this .nycrc.json configuration to make sure that the coverage it collect coverage from  all files in src
```json 
{
  "include": ["src/**/*.ts"],
  "exclude": ["node_modules/**", "coverage/**"],
  "all": true
}
```

munezerobagira commented on src/dummy.ts:
```diff
@@ -1,15 +1,15 @@
-    }
-}
-export default Calculator
+class Calculator{
+    add(a:number, b:number): number{
+        return a+b;
+    }
+    substract(a:number, b:number): number{
+        return a-b;
+    }
+    multiply(a:number, b:number): number{
+        return a*b;
+    }
+    divide(a:number, b:number): number{
+        return a/b;
+    }
+}
+export default Calculator
```
we don't need this dummy you can delete it

munezerobagira replied:
we can use simple test as
expect(true).to.be(true)
all pass

munezerobagira commented on src/math.ts:
we can also remove this files

munezerobagira commented on test/dummy.test.ts:
```diff
@@ -1,45 +1,45 @@
+        //assert
+        expect(result).to.equal(5);
+    })
+    it("should return subtract", ()=>{
+        //arrange
+        const calc = new Calculator();
+
+        // act
+        const result = calc.substract(5,3);
+
+        //assert
+        expect(result).to.equal(2);
+    })
+    it("should return product", ()=>{
+        //arrange
+        const calc = new Calculator();
+
+        // act
+        const result = calc.multiply(2,3);
+
+        //assert
+        expect(result).to.equal(6);
+    })
+    it("should return quotient", ()=>{
+        //arrange
+        const calc = new Calculator();
+
+        // act
+        const result = calc.divide(3,3);
+
+        //assert
+        expect(result).to.equal(1);
+    })
```
```suggestion
```

munezerobagira commented on test/math.test.ts:
```diff
@@ -0,0 +1,20 @@
+  it("should return 10 when have  8 + 2", () => {
+    expect(sum(8, 2)).to.equal(10);
+  });
+  it("Should return the 6 when we have 8 - 2", () => {
+    expect(sub(8, 2)).to.equal(6);
+  });
+  it("should return 16 when we have 8 * 2", () => {
+    expect(multiply(8, 2)).to.equal(16);
+  });
+  it("Should return 4 when we have 8 / 2", () => {
+    expect(div(8, 2)).to.equal(4);
+  });
```
```suggestion

```
lets delete this math.test  its irrevant

Bananayosostene replied:
file was deleted

munezerobagira commented on src/math.ts:
remove this file also

munezerobagira commented on test/math.test.ts:
```diff
@@ -0,0 +1,20 @@
+
+describe("Testing Index", () => {
+  it("should return 10 when have  8 + 2", () => {
+    expect(sum(8, 2)).to.equal(10);
```
```suggestion
    expect(8+2).to.equal(10);
```
Can be as simple as that

munezerobagira commented on test/math.test.ts:
```diff
@@ -0,0 +1,20 @@
+    expect(sum(8, 2)).to.equal(10);
+  });
+  it("Should return the 6  when we have 8 - 2", () => {
+    expect(sub(8, 2)).to.equal(6);
```
```suggestion
    expect(8-2).to.equal(6);
```

munezerobagira commented on test/math.test.ts:
```diff
@@ -0,0 +1,20 @@
+  it("Should return 4 when we have 8 / 2", () => {
+    expect(div(8, 2)).to.equal(4);
+  });
+});
```
```suggestion
});
```

Bananayosostene committed:
feat(coverage): integrate coveralls code coverage service
added coveralls code coverage service
implemented coverage reporting for better code quality
[Delivers #187364719]
